### DISCUSSION TITLE
GrandOrgue for Raspberry pi (how to reduce sample set size?)

### DISCUSSION THREAD

**Discussion by thomz92** [GrandOrgue/grandorgue#992] [Help]
Hi,

I'm looking for ways to reduce the Freisach sample set by Piotr Grabowski so that it's playable on my Raspberry Pi 4 (4gb ram). I'm new to a lot of this audio stuff.

1. My first thought in my naivety was - convert all the samples to 196kbs mp3, but I guess that's not a supported format? Is .wav the only supported format?
2. My second thought was to use ffmpeg to reduce the sound quality to match that of a 196kbs mp3. I'm not exactly sure how to do this?
3. My third thought was "why is the demo sample set so tiny?". I look at the encoder information and it says "wavepack (WVPK). But I read up on this and these files are usually .wv file extensions, but all the demo sample files are .wav. How does this work?
4. My fourth thought was "why are 90% of the sample notes missing in this demo set?". The demo set only has octaves of F# and G. Looking at the .organ definition file, from what i can understand, these notes are loaded and the pitch is adjusted by GrandOrgue. Could i use this method?

My goal is not authentic reproduction of the Friesach organ, but to get all or most of the sound quality with 2.5 - 3gb or so of spare ram. Any help would be appreciated!

Thanks,
Tom E.

**Comment by midimusic** [comment]
The Freisach sampleset needs well over 8GB to work properly so whatever compression you do it will never work in 4GB.

The demo sampleset was specifically made to be small so that it could be distributed with GO without making the package excessively large and to be sure it would work with limited amounts of memory. The samples are WAVEPACKED to keep the file sizes small but they are expanded again when loaded so WAVEPACKING does not reduce the memory requirement.

If you want an easy life I suggest you use an organ which will work in within 4GB memory. (There are not many). Alternatively you can edit the samples to make their duration shorter e.g. reduce 5 second samples down to 2 seconds. You will have to re-loop them. Another option is to use fewer notes per octave e.g. 3 or 4 instead of 12 - to do this you will need to be proficient in editing ODF's.

csw900

**Reply by thomz92** [reply]
Thanks csw900. I haven't found one that I'm happy with within the 4gb range. Theoretically then I could reduce the Friesach organ requirements from ~8gb to ~2gb if i used 3 notes per octave instead of 12?

**Comment by ttsirkia** [comment]
Are you really sure that Raspberry PI is capable of running GO properly? With any more than just one note? If yes, that is quite amazing.

I could recommend this organ: https://piotrgrabowski.pl/dluga-koscielna/ Although my computer has enough memory, I like more the acoustics of this instrument compared to Friesach. The memory requirements for this organ might fit Raspberry PI.

**Reply by thomz92** [reply]
I have the memory limit set to 3560 or so, 16 bit samples, single attack/release, lossless compression - but I tried several by Piotr but the only one I was able to load fully was Ledziny: 
https://piotrgrabowski.pl/ledziny-st-clement/. It kept giving me the "out of memory, organ not fully loaded" error.

The raspberry pi 4 cpu seems capable - I had no problem with polyphony (I'm no amazing organist, but 5-6 notes at a time playing tutti on Ledziny, it had no problem. I had task manager pulled up at one point and i think i remember cpu usage hovering at around 25% during normal playing.

**Reply by ttsirkia** [reply]
Hmm, maybe GO is then more memory-intensive than CPU-intensive. I've thought that it would require more CPU power than GO is able to provide.

With my settings, Długa Kościelna takes less than 4 GB. Loseless compression is not checked, 24 bits, All loops, Single attack and Single release. If you try with the same settings but just 16 bits, is it really so that the memory is not enough?

**Reply by ttsirkia** [reply]
I tried my settings with memory limit of 3560 and the load was succesful.

**Reply by ttsirkia** [reply]
Here is a user having Friesach in Raspberry Pi: https://github.com/GrandOrgue/grandorgue/discussions/958

**Reply by thomz92** [reply]
I would guess he has the 8gb raspberry pi 4, not the 4gb. I will probably try Długa Kościelna or some other organs again, because I looked into editing ODFs and its pretty daunting. Thanks for the help!

**Reply by lbssousa** [reply]
I currently run GrandOrgue on a Raspberry Pi 4 with 8GB of RAM, but I've been able to run it with some simpler organ samplesets (Green Positiv and Ledziny St. Clement by Piotr Grabowski; St. Maria Nordheim by PROSPECTUM) on a Raspberry Pi 3 with only 1GB of RAM. **The [64-bit version of Raspberry Pi OS](https://downloads.raspberrypi.org/raspios_arm64/images/raspios_arm64-2022-01-28/2022-01-28-raspios-bullseye-arm64.zip) may be needed for loading samplesets which require more than 3GB of RAM**.

The largest sampleset by Piotr Grabowski I was able to load on my RPi4 without enabling lossless compression is Azzio, but, by enabling lossless compression, I was able to load Giubiasco, Melcer and Lipiny as well.

Currently I use the following samplesets with my RPi4:
* By Piotr Grabowski
  * [Green Positiv](https://piotrgrabowski.pl/green-positiv/)
  * [Ledziny St. Clement](https://piotrgrabowski.pl/ledziny-st-clement/)
  * [Azzio](https://piotrgrabowski.pl/azzio/)
* By PROSPECTUM
  * [St. Maria Nordheim](https://www.prospectum.com/index.php?language=english&id1=2&id2=10) (Hauptwerk package) + [ODF](https://mps-orgelseite.de/home/filebase/index.php?file/40-odf-st-maria-nordheim/)
* By Sonus Paradisi
  * [Chapelet Spanish Collection Vol. 1](https://www.sonusparadisi.cz/en/chemnitz-sauer-surround-sample-set-409.html) (Hauptwerk package) + [ODF](https://mps-orgelseite.de/home/filebase/index.php?file/76-go-odf-for-the-free-sample-set-by-sonus-paradisi-of-the-galician-organ-from-the/)

**Reply by thomz92** [reply]
Thanks for those as well. I'd have to check, but I don't think Raspberry Pi OS 64bit is stable enough for my purposes (I use it as a full time desktop computer). I have a couple questions out of curiosity:

- Is an mp3 sample set unheard of for a VST or virtual instrument like Grand Orgue? It seems like 192kbs mp3 would be fine for most people, and it sure would make these sample sets more lightweight. I like the massive sound of a big pipe organ in a cathedral. If it's less than CD quality audio, I wouldn't suffer.
- For the raspberry pi, have you tried and do you recommend dry sample sets with convolution reverb?

**Reply by lbssousa** [reply]
> (...) I'd have to check, but I don't think Raspberry Pi OS 64bit is stable enough for my purposes (I use it as a full time desktop computer) (...)

For this case, I would recommend to use another microSD card with Raspberry Pi OS 64-bit only for running GrandOrgue, while keeping your current general-purpose microSD card untouched.

**Reply by thomz92** [reply]
That's an option I can consider if I need to, but at the moment I'm stuck with not enough ram for the organ sample set I want :)

**Reply by ahall41** [reply]
You can reduce the number of stops which are loaded - just load a few stops from each manual to start with, and then build up.  Some stops will use more memory than others. Friesach takes about 6.5Gb for a full set, so you would have room for about 35-40% of the available stops.

**Reply by thomz92** [reply]
That sounds good! How do you control which stops are loaded? Right now it just quits when it runs out of memory and says "some parts are not loaded". I have no idea which parts.

**Reply by ahall41** [reply]
From Audio/Midi check Organ Settings. Here, amongst many things, you can choose which windchest groups and _ranks_ you wish to load. For example, I have the Oloron sample set with but only 8Gb of RAM, so I've only loaded one of the channels.
Start by collapsing the tree and then set the default to under Sample Channels to Don't Load. Then go through the ranks you would like, and change to Stereo.

**Reply by thomz92** [reply]
That sounds good, I'll try that! Thanks!
